typescript/src/Common.ts

```diff
@@ -67,6 +67,24 @@ function prepareIt(): void {
             }
         });
     }
+
+    $("input[required]").bind("invalid", (e) => {
+        const el = e.currentTarget;
+        const label = $(`label[for="${el.getAttribute("id")}"]`);
+        const span = document.createElement("span");
+        $(el).addClass("error");
+        $(span).attr("role", "alert");
+        $(span).addClass("sr-only error");
+        $(span).text(`${label.text()} required`);
+        label.append(span);
+    });
+
+    $("input[required]").on("change", (e) => {
```

Comment:
The `change` event bubbles, so this would be better as a delegated handler, like this:

```javascript
$(document).on("change", "input[required]", (e) => {
  // ...
});
```

This has the benefit of working for any `input` with a `required` attribute, regardless of when that element is placed in the DOM. Doing it as-is will work for anything in the DOM once the `ready` event triggers, but will not work for any dynamically added elements.